Text wrapping fails on FreeBSD, PyQt5

From mailing list:
> 
> I'm a maintainer of the FreeBSD port of TraitsUI (i.e. a source package for TraitsUI in the FreeBSD ports system). With the help of more experienced port maintainers, during the update to version 6.1.2 the TraitsUI port obtained an ability to run its tests. Most of the tests pass, some skip, and one fails. I want to report the failure in a hope that somebody might be interested.
> 
>    Tested system:
> traitsui 6.1.2 + Qt5

```
$ uname -a
FreeBSD vovaasus 12.0-RELEASE FreeBSD 12.0-RELEASE r341666 GENERIC  amd64

 Copy-pasted end of the tests output:

test_nonmodal_view_application_ok (traitsui.tests.test_view_application.TestViewApplication) ... ok
traitsui.tests.test_visible_when_layout.test_visible_when_layout ... ok

----------------------------------------------------------------------
Ran 161 tests in 11.772s

OK (SKIP=13)
Coverage.py warning: No data was collected. (no-data-collected)
cd /usr/ports/graphics/py-traitsui/work-py36/traitsui-6.1.2 && /usr/bin/env ETS_TOOLKIT=qt4 QT_API=pyqt5 /usr/local/bin/python3.6 -m coverage run -p -m nose.core -v traitsui.qt4.tests
test_wrap_text_basic (traitsui.qt4.tests.test_helper.TestWrapText) ... ok
test_wrap_text_empty (traitsui.qt4.tests.test_helper.TestWrapText) ... ok
test_wrap_text_narrow (traitsui.qt4.tests.test_helper.TestWrapText) ... ok
test_wrap_text_narrow_short (traitsui.qt4.tests.test_helper.TestWrapText) ... FAIL
test_wrap_text_short (traitsui.qt4.tests.test_helper.TestWrapText) ... ok
test_panel_has_toolbar_buttons_qt4 (traitsui.qt4.tests.test_ui_panel.TestUIPanel) ... ok
test_subpanel_has_toolbar_no_buttons_qt4 (traitsui.qt4.tests.test_ui_panel.TestUIPanel) ... ok
test_subpanel_no_toolbar_nor_button_in_widget (traitsui.qt4.tests.test_ui_panel.TestUIPanel) ... ok

======================================================================
FAIL: test_wrap_text_narrow_short (traitsui.qt4.tests.test_helper.TestWrapText)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/traitsui/qt4/tests/test_helper.py", line 105, in test_wrap_text_narrow_short
    self.assertEqual(lines[19][-1], '\u2026')
AssertionError: '.' != '…'
- .
+ …


----------------------------------------------------------------------
Ran 8 tests in 0.206s

FAILED (failures=1)
Coverage.py warning: No data was collected. (no-data-collected)
*** Error code 1

Stop.
make: stopped in /usr/ports/graphics/py-traitsui
```

--- a/traitsui/qt4/tests/test_helper.py
+++ b/traitsui/qt4/tests/test_helper.py
@@ -30,6 +30,7 @@ lorem_ipsum = (
     "officia deserunt mollit anim id est laborum."
 )
 
+
 def get_expected_lines(text, width):
     expected_lines = []
     for paragraph in text.splitlines():
@@ -101,8 +102,13 @@ class TestWrapText(unittest.TestCase):
         # add one char slack as depends on OS, exact font, etc.
         self.assertTrue(all(len(line) <= 21 for line in lines))
         # different os elide the last line slightly differently,
-        # just check last character shows elision.
-        self.assertEqual(lines[19][-1], '\u2026')
+        # just check end of last line shows elision.
+        # In most systems elision is marked with ellipsis
+        # but it has been reported as "..." on NetBSD.
+        if lines[19][-1] == '.':
+            self.assertEqual(lines[19][-3:], '...')
+        else:
+            self.assertEqual(lines[19][-1], '\u2026')
 
     def test_wrap_text_short(self):
         font = create_traitsfont("Courier")

